### PR TITLE
support multiple response types per client

### LINKED ISSUE #270: Set allowed response_types by grant.
In Identity Server there is the ability to have a client support multiple response_type's through setting an Allowed Grant (Implicit, Hybrid, ClientCredentials). This allows a Client to support both `id_token` and `id_token token`.

This is functionality that is used by `oidc-client-js` to get a simplified session status response.

Would this be something `django-oidc-provider` would support? or is there a workaround to allow 1 client 2 response_types?

### PR DESCRIPTION
**Pull request by q3aiml**
@juanifioren What do you think about adding support for multiple response types per client?

The Dynamic Client Registration spec specifies multiple response_types
and grant_types per client
(https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata).
Since grant_types can be inferred from response_types we should be able
to support both without needing to store grant_types.

This also helps with oidc-client-js which expects a client that supports
both "id_token" and "id_token token".

On the admin page the select box changes to allow selecting multiple response types:
image

And on the list page it shows the description, similar to before, delimited with commas if more than one response type is selected:
image

Closes #270

### CODE REVIEW

File: oidc_provider/tests/app/utils.py [hunk 1]
```diff
@@ -58,12 +59,16 @@ def create_fake_client(response_type, is_public=False, require_consent=True):
         client.client_secret = ''
     else:
         client.client_secret = str(random.randint(1, 999999)).zfill(6)
-    client.response_type = response_type
     client.redirect_uris = ['http://example.com/']
     client.require_consent = require_consent
 
     client.save()
 
+    if isinstance(response_type, ("".__class__, u"".__class__)):
```
Comment (juanifioren) [inline oidc_provider/tests/app/utils.py]: What is this for? we need more inline comments :+1:

File: oidc_provider/admin.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ class ClientAdmin(admin.ModelAdmin):
     fieldsets = [
         [_(u''), {
             'fields': (
-                'name', 'owner', 'client_type', 'response_type', '_redirect_uris', 'jwt_alg',
+                'name', 'owner', 'client_type', 'response_types', '_redirect_uris', 'jwt_alg',
```
Comment (juanifioren) [inline oidc_provider/admin.py]: How its presented to the user on admin? via select? raw id?

Comment (q3aiml) [reply oidc_provider/admin.py]: It's using the multiselect mechanism that is built in to django for many to many relationships:
image

On the admin list page it then show the descriptions similar to before:
image

Good question. I'll also update the PR description with this.

**Comment by q3aiml** [comment]
Good catch on the docs. That should be updated now.

I do find the many to many relationship not quite as clean as a single field, especially when looking at that client creation example, but I haven't found a better option. Keeping response_types as a single field requires more code or a new dependency such as https://github.com/goinnn/django-multiselectfield to maintain nice admin functionality. I am unsure if that is better. Let me know if you have a preference.

**Comment by juanifioren** [comment]
Multiple choice select it's OK for me

**Comment by q3aiml** [comment]
Updated the Properties section in relyingparties.rst as well as a tiny update in accesstokens.rst

**Pull request merged**